scripts/validate_documentation.py

```diff
@@ -0,0 +1,227 @@
+import argparse
+from fnmatch import fnmatch
+import json
+import os
+import pickle
+import re
+import sys
+
+import docutils.nodes
+
+DOCUMENTATION_SOURCE = os.path.join(os.curdir, '../doc/source')
+DOCTREE_PATH = '../doc/build/doctrees/{}.doctree'
+RST_PATH = '../doc/source/{}.rst'
+
+ERROR_MSGS = {
+    'DT01': "Found bullet list within block quote. Use 0 spaces for top-level "
+            "list and 2 spaces for sub-lists",
+    'WS01': "Indentation uses tabulation",
+    'WS02': "Trailing whitespaces",
+}
+
+STARTING_WHITESPACE_RE = re.compile(r'^(\s+).*\n$', re.MULTILINE)
+TRAILING_WHITESPACE_RE = re.compile(r'^.*([\t ]+)\n$', re.MULTILINE)
+
+
+class DocumentChecker(object):
+
```

Comment:
May be a short docstring for the class? If you explain what the script does at the beginning of the file there is not much to say here, but a comment on what does this class do, or how is it expected to be used, can be useful.